pkg/adapter/containers.go

```diff
@@ -784,5 +786,71 @@ func (r *LocalRuntime) Top(cli *cliconfig.TopValues) ([]string, error) {
 	if err != nil {
 		return nil, errors.Wrapf(err, "unable to lookup requested container")
 	}
-	return container.Top(descriptors)
+
+	// Also support comma-separated input.
+	psgoDescriptors := []string{}
+	for _, d := range descriptors {
+		for _, s := range strings.Split(d, ",") {
+			if s != "" {
+				psgoDescriptors = append(psgoDescriptors, s)
+			}
+		}
+	}
+	output, err := container.Top(psgoDescriptors)
+
+	// If we encountered an ErrUnknownDescriptor error, fallback to executing
+	// ps(1) in the container. This ensures backwards compatibility to users
+	// depending on ps(1) and makes sure we're ~compatible with docker.
+	if errors.Cause(err) != psgo.ErrUnknownDescriptor {
+		return output, err
+	}
+
+	// First try to execute ps(1) in the container but with the initial descriptors.
+	output, execPSError := r.execPS(container, descriptors)
+	if execPSError == nil {
+		return output, nil
+	}
+
+	// Some images (e.g., fedora) don't ship with ps(1), so we need to check if
+	// ps(1) is present at all. If it's present, then return the error of the
+	// previous call. Otherwise, don't bother an call psgo without a descriptor
+	// which is equivalent to a `ps -ef` call.
+	if _, err := r.execPS(container, []string{}); err == nil {
```

Comment:
Shouldn't we return the original err now, IE psgo.ErrUnknownDescriptor?
Seems we are checking if the ps command fails and their is no ps in the container then we should just return psgo.ErrUnknownDescriptor

Reply:
Makes sense, thanks!